include/deal.II/fe/fe_tools.templates.h

```diff
@@ -103,23 +98,62 @@ namespace FETools
             break;
           }
 
+      dealii::internal::GenericDoFsPerObject dpo;
+
+      std::vector<dealii::internal::GenericDoFsPerObject> dpos_in(fes.size());
+
       for (unsigned int i = 0; i < fes.size(); ++i)
         if (multiplicities[i] > 0)
-          {
-            // TODO: the implementation makes the assumption that all faces have
-            // the same number of dofs -> don't construct DPO but
-            // PrecomputedFiniteElementData
-            AssertDimension(fes[i]->n_unique_quads(), 1);
-
-            multiplied_dofs_per_vertex +=
-              fes[i]->n_dofs_per_vertex() * multiplicities[i];
-            multiplied_dofs_per_line +=
-              fes[i]->n_dofs_per_line() * multiplicities[i];
-            multiplied_dofs_per_quad +=
-              fes[i]->n_dofs_per_quad(0) * multiplicities[i];
-            multiplied_dofs_per_hex +=
-              fes[i]->n_dofs_per_hex() * multiplicities[i];
+          dpos_in[i] =
+            dealii::internal::GenericDoFsPerObject::generate(*fes[i]);
+
+      const auto initialize =
```

Comment:
This part is also pretty hard to understand compared to the old version. Could you add a sentence or two here explaining how we set up each field in `dpo`?

Reply:
I have added some comments.